### PR TITLE
SVCCA experiments

### PR DESCRIPTION
**Pull request by ajlee21**
This PR accomplishes the following:

1.  Minor organization updates including: moving config files into scripts/ dir, updating the readme to add more details about the difference between experiment 0 vs 1

2.  Add scripts for experiment 1, which experiments with the SVCCA metric.  The scripts in experiment 1 are the same as experiment 0 except:
  test_svcca_and_dimensionality.ipynb:  tests the affects of the input dimensionality on svcca score
  test_svcca_and_transformations.ipynb:  tests the affect of affine transformations on svcca score 
  2_simulate_data_truncated.ipynb:  modified simulation script to allow for subsampling of genes since it appears that fewer dimensions yields more expected results (i.e. svcca score of input vs itself is closer to 1 and svcca score of input vs permuted input is reduced).  

Based on this analysis, some next steps are to look into and modify the definition of batch effect and to try to determine if svcca is an appropriate metric to use for this application.  Any thoughts or suggestions are very welcome

### CODE REVIEW

File: scripts/experiment_1/functions/cca_core.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
+# Copyright 2016 Google Inc.
```
Comment (ypar) [inline scripts/experiment_1/functions/cca_core.py]: Is it really necessary to keep copy pasting this script? Why not just create a bin where all external scripts are located and call from it? You seem to have at least two copies of this script.

File: scripts/experiment_1/functions/vae.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        latent_dim,
+        epsilon_std,
+        rnaseq,
+        base_dir,
+        analysis_name):
+    """
+    Train 2-layer Tybalt model using input dataset
+
+    Arguments:
+    -- learning_rate: step size used for gradient descent. In other words, its how quickly the  methods is learning
```
Comment (ypar) [inline scripts/experiment_1/functions/vae.py]: its -> it's

File: scripts/experiment_1/functions/vae.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    fig = ax.get_figure()
+    fig.savefig(hist_plot_file)
+
+    del ax, fig
+
+    # Output
+
+    # Save training performance
+    history_df = pd.DataFrame(hist.history)
+    history_df = history_df.assign(learning_rate=learning_rate)
```
Comment (ypar) [inline scripts/experiment_1/functions/vae.py]: Since they are not doing any new operations, they can just be added rather than using the assign function

File: scripts/experiment_1/functions/vae.py [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+    # build a generator that can sample from the learned distribution
+    # can generate from any sampled z vector
+    decoder_input = Input(shape=(latent_dim, ))
+    x_decoded_mean = decoder_model(decoder_input)
+    decoder = Model(decoder_input, x_decoded_mean)
+    weights = []
+    for layer in decoder.layers:
+        weights.append(layer.get_weights())
+
+    # Multiply hidden layers together to obtain a single representation of gene weights
```
Comment (ypar) [inline scripts/experiment_1/functions/vae.py]: Why are these commented out? Would they be deleted eventually?

File: scripts/experiment_1/nbconverted/1_train_vae.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+             os.path.join(base_dir, "output", "stats"),
+             os.path.join(base_dir, "output", "viz")
+             ]
+
+# Check if analysis directory exist otherwise create
+
+for each_dir in base_dirs:
+
+    if os.path.exists(each_dir):
+        print('directory already exists: {}'.format(each_dir))
```
Comment (ypar) [inline scripts/experiment_1/nbconverted/1_train_vae.py]: With newer python3, you can just call 

`print(f' dir exists {each_dir}.')`

File: scripts/experiment_1/nbconverted/1_train_vae.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    "data",
+    "input",
+    "train_set_normalized.pcl")
+
+
+# In[6]:
+
+
+# Read data
+normalized_data = pd.read_table(
```
Comment (ypar) [inline scripts/experiment_1/nbconverted/1_train_vae.py]: I am noticing that they are always transposed. Is there any reason why they are not saved in a transposed form?

File: scripts/experiment_1/nbconverted/1_train_vae.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+             os.path.join(base_dir, "output", "viz")
+             ]
+
+# Check if analysis directory exist otherwise create
+
+for each_dir in output_dirs:
+    new_dir = os.path.join(each_dir, train_architecture)
+    
+    if os.path.exists(new_dir):
+        print('directory already exists: {}'.format(new_dir))
```
Comment (ypar) [inline scripts/experiment_1/nbconverted/1_train_vae.py]: Same comment from above for py3 print formatting

File: scripts/experiment_1/nbconverted/1_train_vae.py [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    else:
+        print('creating new directory: {}'.format(new_dir))
+    os.makedirs(new_dir, exist_ok=True)
+
+
+# In[9]:
+
+
+# Train nonlinear (VAE)
+vae.tybalt_2layer_model(learning_rate,
```
Comment (ypar) [inline scripts/experiment_1/nbconverted/1_train_vae.py]: Since they are used only once, they could just be used instead of being assigned as new variables. e.g.

```
vae.tybalt_2layer_model(d['learning_rate'], d['batch_size'], d['epochs'], d['kappa'], d['intermediate_dim'], d['latent_dim'],d['epsilon_std'], ...)
```

File: scripts/experiment_1/nbconverted/2_simulate_data_truncated.py [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+
+# Create directories
+base_dir = os.path.abspath(os.path.join(os.getcwd(),"../.."))
+
+new_dir = os.path.join(base_dir, "data", "simulated")
+
+analysis_dir = os.path.join(new_dir, analysis_name)
+
+if os.path.exists(analysis_dir):
+    print('directory already exists: {}'.format(analysis_dir))
```
Comment (ypar) [inline scripts/experiment_1/nbconverted/2_simulate_data_truncated.py]: could use simpler print formatting

File: scripts/experiment_1/nbconverted/2_simulate_data_truncated.py [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+else:
+    print('creating new directory: {}'.format(analysis_dir))
```
Comment (ypar) [inline scripts/experiment_1/nbconverted/2_simulate_data_truncated.py]: could use simpler print formatting

File: scripts/experiment_1/functions/cca_core.py (hunk 1 above)
Comment (ajlee21) [reply scripts/experiment_1/functions/cca_core.py]: Very good point.  I have modified the directories accordingly

File: scripts/experiment_1/functions/vae.py (hunk 2 above)
Comment (ajlee21) [reply scripts/experiment_1/functions/vae.py]: fixed.  At some point I will want to update the text in these descriptions to be cleaner

File: scripts/experiment_1/nbconverted/1_train_vae.py (hunk 6 above)
Comment (ajlee21) [reply scripts/experiment_1/nbconverted/1_train_vae.py]: This is the dimensions of the downloaded data.  So each time it is read it it gets transposed.  I understand that I would be doing this each time I read in the data, I think the rationale here is that I wanted to make sure people knew that this was the raw data that was downloaded and that the only change was that it was transposed

File: scripts/experiment_1/nbconverted/1_train_vae.py (hunk 5 above)
Comment (ajlee21) [reply scripts/experiment_1/nbconverted/1_train_vae.py]: I am using py3.5, I don't think this syntax is supported until py3.6  Getting an error so just leaving it in its original format for now.
